src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/_params.py

```diff
@@ -4,15 +4,31 @@
 #---------------------------------------------------------------------------------------------
 #pylint: disable=line-too-long
 
-from azure.cli.core.commands.parameters import (location_type, get_enum_choices)
+from azure.cli.core.commands.parameters import (location_type, get_enum_choices, get_resource_name_completion_list)
 from azure.cli.core.commands import register_cli_argument
-
 from azure.mgmt.iothub.models.iot_hub_client_enums import IotHubSku
+from ._factory import iot_hub_service_factory
+from .custom import iot_device_list
+
+
+def get_device_id_completion_list():
+    def completer(prefix, action, parsed_args, **kwargs):#pylint: disable=unused-argument
+        client = iot_hub_service_factory(kwargs)
+        if parsed_args.hub_name:
+            return [d.device_id for d in iot_device_list(client, parsed_args.hub_name, top=100)]
+        else:
+            return []
+    return completer
 
 
+register_cli_argument('iot', 'hub_name', options_list=('--hub-name', '--name', '--hub', '-n'),
```

Comment:
Okay.

I think @tjprescott raised similar comments to me as in #953, it says "Use consistent names (hub_name instead of hub and name)" and in https://github.com/Azure/azure-cli/pull/951#discussion_r80070944, it says "Otherwise you'll have multiple entries doing the same thing because you have to alias both 'hub' and 'name' (and since name can refer to different things in different contexts, it can get confusing)"